### DISCUSSION TITLE
KeyError: 'agg_matches'

### DISCUSSION THREAD

**Discussion by Sinistr0** [jertel/elastalert2#957] [Q&A]
Hello, I have an error when trying to execute a rule. below are the rule settings and the error.
yaml:

` alert:
  - "post2"
description: "xxxx - xxxxxxxxxxxxxx"
filter:
- query:
     query_string:
       query: "_exists_: TMCMLogDetectedHost"
filter:
- query:
     query_string:
       query: "_exists_: deviceCustomString5"

generate_kibana_discover_url: false
http_post2_ignore_ssl_errors: true

http_post2_timeout: 30
http_post2_url: "https://xxxxxxxxxx/api/v1/hooks/webhook_xxxxxxxxxxxxxxx3"
http_post2_payload:
  nomedaregra: xxxxx - xxxxxxxxxx
  ActionResultorRiskLevel: "{{deviceCustomString5}}"
  fileHash: "{{fileHash}}"
  filePath: "{{filePath}}"
  TMCMLogTarget: "{{TMCMLogTarget}}"
  sourceProcessName: "{{sourceProcessName}}"
  name: "{{name}}"
  id: "{{_id}}"
  data: "{{event_time}}"
  deviceReceiptTime: "{{deviceReceiptTime}}"
  severity: "{{severity}}"
  TMCMLogDetectedHost: "{{TMCMLogDetectedHost}}"
  cliente: "{{cliente}}"
  açaotomada: "{{deviceAction}}"
  HostName: "{{deviceHostName}}"
  deviceAction: "{{deviceAction}}"

index: "xxxxxxxx*"
is_enabled: false
name: "xxx - xxxxxxxxxxxxxxxxxxx"
owner: "xxxxx"
realert:
  minutes: 0
timestamp_field: "@timestamp"
timestamp_type: "iso"
type: cardinality
cardinality_field: "TMCMLogDetectedHost.keyword"
max_cardinality: 3
timeframe:
  hours: 24
#use_strftime_index: false `

and here's the error:
` user:/opt/elastalert# elastalert-test-rule rules/xxxx_-xxxxxxxxxxxxxxxx.yaml --days 60 --alert
Got 36 hits from the last 60 days ` 

` Available terms in first hit: 
        deviceReceiptTime
        syslog
        severity
        @timestamp
        deviceCustomString3
        TMCMLogDetectedIP
        deviceHostName
        deviceProduct
        deviceAction
        deviceCustomString4Label
        deviceCustomString3Label
        destinationNtDomain
        cefVersion
        name
        deviceCustomString5
        sourceAddress
        deviceVersion
        event_time
        deviceVendor
        deviceEventClassId
        sourceProcessName
        deviceCustomString4
        deviceCustomString5Label
        TMCMLogTarget
        cliente
        TMCMLogDetectedHost
        devicePayloadId
        @version
        ApexCentralHost
        deviceNtDomain
        port
        sourceHostName
        tags
        hostname
        deviceFacility `

` INFO:elastalert:1 rules loaded
Traceback (most recent call last):
  File "/usr/local/bin/elastalert-test-rule", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/elastalert/test_rule.py", line 502, in main
    test_instance.run_rule_test()
  File "/usr/local/lib/python3.10/site-packages/elastalert/test_rule.py", line 492, in run_rule_test
    self.run_elastalert(rule_yaml, conf)
  File "/usr/local/lib/python3.10/site-packages/elastalert/test_rule.py", line 408, in run_elastalert
    client.run_rule(rule, self.endtime, self.starttime)
  File "/usr/local/lib/python3.10/site-packages/elastalert/elastalert.py", line 821, in run_rule
    for x in range(len(rule['agg_matches'])): 
KeyError: 'agg_matches' `

am i doing something wrong?

**Comment by jertel** [comment]
I'm having trouble reading this message. Please see #11 and ensure all the requested data is present and the message is formatted properly.
